BDArmory/Control/MissileFire.cs

```diff
@@ -5525,7 +5502,8 @@ bool SmartPickWeapon_EngagementEnvelope(TargetInfo target)
                                     }
                                     if (distance < ((EngageableWeapon)item.Current).engageRangeMin || firedMissiles >= maxMissilesOnTarget)
                                         candidateTDPS *= -1f; // if within min range, negatively weight weapon - allows weapon to still be selected if all others lost/out of ammo
-
+                                    if (unguidedWeapon && distance > ((EngageableWeapon)item.Current).engageRangeMax / 10)
+                                        candidateTDPS *= -1f; // don't fire unguided ordinance from long range
```

Comment:
This combined with line 5504 could potentially make `candidateTDPS` positive again if `engageRangeMax / 10 < engageRangeMin`.
Adding a check for `candidateTDPS` being positive should avoid that though. Alternatively, using `max(engageRangeMin, engageRangeMax / 10)` would also work, but maybe isn't as simple?

Similarly for the other `candidateTDPS *= -1f;` and `candidateYield *= -1f;` cases below.

However, the check on line 6490 should mean that the code never gets to these lines under those conditions.

Reply:
Yeah, looks like there's some dueling logic that was missed first pass; I've made some changes that should fix things.